### DISCUSSION TITLE
Is OpenVINO really accelerates Smart Search job?

### DISCUSSION THREAD

**Discussion by DX37** [immich-app/immich#8104] [General]
I've been testing OpenVINO a while. Not sure that it accelerating anything. On the contrary, it uses more threads in gunicorn and also uses render from GPU, while providing same results in seconds per jobs.

Here's what I've got with Smart Search job on Intel UHD Graphics 630 in i5-10500.

OpenVINO, 2 concurrent jobs, ~2 assets per 5 seconds
![11](https://github.com/immich-app/immich/assets/12829428/c792a12d-5261-4ce2-88d5-37e7aa15a206)

OpenVINO, 4 concurrent jobs, ~4 assets per 9 seconds
![22](https://github.com/immich-app/immich/assets/12829428/79a853e9-78d8-4ec2-8306-e909a9d4c997)

And here's results with CPU, but with the same time.

CPU, 2 concurrent jobs, ~2 assets per 5 seconds
![33](https://github.com/immich-app/immich/assets/12829428/e5c6feba-b74c-4f07-8fd9-e0cc786b8f0f)

CPU, 4 concurrent jobs, ~4 assets per 9 seconds
![44](https://github.com/immich-app/immich/assets/12829428/f5a93fc4-fe3a-474b-9562-47955cad9aa2)

So, is it me doing something wrong, or is it OpenVINO that is supposed to act like this, or is it something else?

**Comment by DX37** [comment]
Or is it model dependent? I'm using `nllb-clip-large-siglip__v1`

**Reply by mertalev** [reply]
It is, to some extent. Certain models can get optimized better than others, etc. But I doubt that's the cause here.

**Comment by DX37** [comment]
I wonder if Immich is still sending assets to CPU provider.
![image](https://github.com/immich-app/immich/assets/12829428/9f2d6106-d3bc-44af-821a-97a6a1eea842)

**Reply by mertalev** [reply]
No, ONNX Runtime decides where parts of the model are allocated. It will put almost everything on the OpenVINO provider except for specific things that are faster to do on CPU.

**Comment by mertalev** [comment]
There are too many factors to say anything decisive, but some thoughts:

- Models are compiled to OpenVINO format on the first load. This can take a few minutes, especially with a large model.
- The GPU utilization looks very spotty. It might not be getting enough work to do.
- Relatedly, inputs to the GPU aren't batched right now. OpenVINO may be more sensitive to this than CUDA.
- The GPU isn't very strong. It might not be powerful enough to make a big difference.
